lib/bap/bap_self.ml

```diff
@@ -151,11 +156,11 @@ module Create() = struct
     let manpage (man:manpage_block list) : unit =
       term_info := Term.info ~doc ~man plugin_name
 
-    type 'a reader = 'a param -> 'a
-    let parse () : 'a reader =
+    type reader = {get : 'a. 'a param -> 'a}
+    let parse configured : unit =
```

Comment:
We should call provided function only after all top level expressions are evaluated, i.e., _after_ the plugin is loaded. 
What we can try to achieve this is to monitor the plugins subsystem events:
```ocaml
let request on_config_ready = 
  Stream.watch Plugins.events (fun subscription -> function
    | `Loaded p when Plugin.name p = plugin_name ->
         on_config_ready {get = (fun p -> Future.peek_exn p)};
         Stream.unsubscribe subscription
    | `Errored name when plugin_name = name -> 
         Stream.unsubscribe subscription)
```

This is not the ideal solution, as it is a little bit fragile, and not very clean, but as an intermediate solution, before we will switch to the global parser, may work.